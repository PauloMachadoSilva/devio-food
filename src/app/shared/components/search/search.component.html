<section class="devio-food-search">
  <form class="">
    <div class="devio-food-search__acao">
      <mat-form-field appearance="outline" [color]="'primary'">
        <mat-label>Pesquisar</mat-label>
        <input #input matInput placeholder="O que vocÃª procura?">
      </mat-form-field>
      <button (click)="consultarProduto(input.value)" mat-fab color="primary" aria-label="Example icon button with a bookmark icon">
        <mat-icon>search</mat-icon>
      </button> 
    </div>
    </form>
</section>