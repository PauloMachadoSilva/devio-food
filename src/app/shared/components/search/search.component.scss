@use './../../../../style/tokens.scss' as *;

.devio-food-search {
    mat-form-field {
        width: 100%;
    }
    &__acao {
        display: flex;
    }
}