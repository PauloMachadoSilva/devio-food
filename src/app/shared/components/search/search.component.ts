import { MatIconModule } from '@angular/material/icon';
import { MatButtonModule } from '@angular/material/button';
import {MatInputModule} from '@angular/material/input';
import { FormsModule, ReactiveFormsModule } from "@angular/forms";
import { Component, EventEmitter, OnInit, Output } from '@angular/core';
import { MatFormFieldModule } from '@angular/material/form-field';

@Component({
  selector: 'devio-food-search',
  templateUrl: './search.component.html',
  styleUrls: ['./search.component.scss'],
  standalone: true,
  imports: [
    FormsModule, 
    MatFormFieldModule, 
    MatInputModule, 
    MatButtonModule,
    MatIconModule

  ],
})
export class SearchComponent {
@Output() itemSearch = new EventEmitter<string>();

  constructor() { }
  
  consultarProduto(event:string) {
    this.itemSearch.emit(event)
  }

}