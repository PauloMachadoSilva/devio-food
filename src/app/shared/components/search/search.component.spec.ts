import { HttpClient, HttpClientModule } from '@angular/common/http';
/* tslint:disable:no-unused-variable */
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { SearchComponent } from './search.component';
import { provideAnimations } from '@angular/platform-browser/animations';
describe('CancelButtonComponent', () => {
  let component: SearchComponent;
  let fixture: ComponentFixture<SearchComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [SearchComponent, HttpClientModule],
      providers: [
        provideAnimations(),
        HttpClient
        
      ]
    }).compileComponents();

    fixture = TestBed.createComponent(SearchComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});