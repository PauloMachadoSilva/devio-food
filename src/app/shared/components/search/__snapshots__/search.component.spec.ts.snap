// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelButtonComponent should create 1`] = `
<div
  id="root0"
>
  <section
    class="devio-food-search"
  >
    <form
      class="example-form ng-untouched ng-pristine ng-valid"
      novalidate=""
    >
      <mat-form-field
        appearance="outline"
        class="mat-mdc-form-field mat-mdc-form-field-type-mat-input mat-form-field-appearance-outline mat-form-field-hide-placeholder mat-primary"
      >
        
        <div
          class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--outlined"
        >
          
          <div
            class="mat-mdc-form-field-flex"
          >
            <div
              class="mdc-notched-outline mdc-notched-outline--upgraded"
              matformfieldnotchedoutline=""
            >
              <div
                class="mdc-notched-outline__leading"
              />
              <div
                class="mdc-notched-outline__notch"
              >
                <label
                  class="mdc-floating-label mat-mdc-floating-label"
                  matformfieldfloatinglabel=""
                  style="transition-duration: 0s;"
                >
                  <mat-label>
                    Pesquisar
                  </mat-label>
                  
                </label>
                
                
                
              </div>
              <div
                class="mdc-notched-outline__trailing"
              />
            </div>
            
            
            
            <div
              class="mat-mdc-form-field-infix"
            >
              
              <input
                aria-invalid="false"
                aria-required="false"
                class="mat-mdc-input-element mat-mdc-form-field-input-control mdc-text-field__input cdk-text-field-autofill-monitored"
                matinput=""
                placeholder="O que vocÃª procura?"
              />
            </div>
            
            
          </div>
          
        </div>
        <div
          class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
        >
          <div
            class="mat-mdc-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
            style="opacity: 1; transform: translateY(0%);"
          >
            
            <div
              class="mat-mdc-form-field-hint-spacer"
            />
          </div>
          
          
        </div>
      </mat-form-field>
    </form>
  </section>
</div>
`;
